TypeScript/src/App.tsx

```diff
@@ -1,46 +1,52 @@
-import React from "react";
+import React, {
+  useState,
+  useMemo,
+  createContext,
+  Dispatch,
+  SetStateAction,
+} from "react";
 import { Route, Routes } from "react-router-dom";
-import { useEffect, useMemo, useState, Dispatch, SetStateAction, createContext } from "react";
-import { initializePage } from "./services/pageInitService";
-
+import { pageContextInterface } from "./models/contextModels.model";
 import Layout from "./ux-elements/Layout/Layout";
 
-interface pageContextInterface {
-  pageNumber: number,
-  setPageNumber?: Dispatch<SetStateAction<number | null | undefined>> 
-}
-
 export const pageContext = createContext<pageContextInterface | null>(null);
 
-function createPageContext(pageNum, setPageNum) {
-  // const [pageNum, setPageNum] = useState<number | null | undefined>(1);
+function createPageContext(
+  pageNum: number,
+  setPageNum: Dispatch<SetStateAction<number>>
+) {
   if (localStorage.getItem("pageId")) {
-    setPageNum(localStorage.getItem("pageId"));
+    setPageNum(+localStorage.getItem("pageId"));
```

Comment:
I think we should check for a diff between the getItem and the pageNum we are setting. Then we can simply setItem if there is a diff and thus less hassle.
`if (pageNum !== localStorage.getItem("pageId")) {
      setPageNum(pageNum);
      localStorage.setItem("pageId", pageNum);
    }`

Reply:
Yeah cleaner for sure

Reply:
changed